parsers/ES_CN.py

```diff
@@ -6,77 +6,60 @@
 
 
 def fetch_consumption(country_code='ES-CN', session=None):
-    if not session:
-        session = Session
+    ses = session or session()
 
     try:
-        response = CanaryIslands(session).get()
+        response = CanaryIslands(ses).get()
     except NoDataException:
-        response = None
+        return None
     except TimestampException:
-        response = None
-
-    if not response:
-        datetime = utcnow().datetime
-        consumption = None
-    else:
-        datetime = get(response.timestamp).datetime
-        consumption = response.demand
+        return None
 
     data = {
         'countryCode': country_code,
-        'datetime': datetime,
-        'consumption': consumption,
+        'datetime': get(response.timestamp).datetime,
+        'consumption': response.demand,
         'source': 'demanda.ree.es'
     }
 
     return data
 
 
 def fetch_production(country_code='ES-CN', session=None):
-
-    if not session:
-        session = Session
+    ses = session or session()
 
     try:
-        response = CanaryIslands(session).get()
+        response = CanaryIslands(ses).get()
     except NoDataException:
-        response = None
+        return None
     except TimestampException:
-        response = None
-
-    if not response:
-        datetime = utcnow().datetime
-    else:
-        datetime = get(response.timestamp).datetime
+        return None
 
     data = {
         'countryCode': country_code,
-        'datetime': datetime,
+        'datetime': get(response.timestamp).datetime,
         'production': {
-          'biomass': 0.0,
-          'coal': 0.0,
-          'nuclear': 0.0,
-          'geothermal': 0.0,
-          'unknown': 0.0
+            'gas': round(response.gas + response.combined, 2),
+            'solar': response.solar,
+            'oil': round(response.vapor + response.diesel, 2),
+            'wind': response.wind,
+            'biomass': 0.0,
+            'coal': 0.0,
+            'nuclear': 0.0,
+            'geothermal': 0.0,
+            'unknown': response.unknown()
         },
         'storage': {},
         'source': 'demanda.ree.es',
     }
 
-    if response:
-        data['production']['gas'] = response.gas + response.combined
-        data['production']['solar'] = response.solar
-        data['production']['oil'] = response.vapor + response.diesel
-        data['production']['wind'] = response.wind
-
-        hidro = response.hydraulic
-        if hidro >= 0.0:
-            data['production']['hydro'] = hidro
-            data['storage']['hydro'] = 0.0
-        else:
-            data['production']['hydro'] = 0.0
-            data['storage']['hydro'] = abs(hidro)
+    hidro = response.hydraulic
```

Comment:
A quick look at El Hierro shows that the hydro actually pumped storage:
https://fr.wikipedia.org/wiki/Centrale_hydro-%C3%A9olienne_d%27El_Hierro

Can you check for the other islands of the Canary Islands?